### PR TITLE
CAKE-5295 | Add test ddb unit & update link with tracking params

### PR DESCRIPTION
**Pull request by JoshuaRogan**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5295

## Description

* Add all params that we know at fetch time. 
* Add ddb test unit
* Slot id code

## Reviewers
@Wikia/cake

### CODE REVIEW

File: app/services/affiliate-slots.js [hunk 1]
```diff
@@ -123,10 +124,39 @@ export default Service.extend({
...
+    const wikiId = this.currentWikiId;
+
+    // fandom_slot_id will be added later
+    const questionMarkOrAmpersan = (unit.link.indexOf('?') > -1) ? '&' : '?';
+    unit.link = `${unit.link}${questionMarkOrAmpersan}fandom_session_id=${session}&fandom_user_id=${userId}&fandom_campaign_id=${unit.category}&fandom_community_id=${wikiId}fandom_page_id=${pageId}&fandom_beacon_id=${beaconId}`;
+    return unit;
+  },
+
+  _createDDBLink(unit, pageId = 'search') {
+    // const beaconId = Cookies.get('wikia_beacon_id');
```
Comment (cschucode) [inline app/services/affiliate-slots.js]: Should this be commented out here?

Comment (JoshuaRogan) [reply app/services/affiliate-slots.js]: Good catch I meant to remove this entire block : https://github.com/Wikia/mobile-wiki/pull/1937/commits/ddec7df2ecfa854a7b01847845ecc212af80ae9c

File: app/services/affiliate-slots.js [hunk 2]
```diff
@@ -123,10 +124,28 @@ export default Service.extend({
...
+  currentUserId: readOnly('currentUser.userId'),
+
+  _updateUnitLink(unit, pageId = 'search') {
+    if (!unit || unit.campaign !== 'ddb') {
+      return unit;
+    }
+
+    const beaconId = Cookies.get('wikia_beacon_id');
+    const session = Cookies.get('wikia_session_id');
+    const userId = this.currentUserId ? this.currentUserId : 'null'; // make sure we actually send null
```
Comment (vforge) [inline app/services/affiliate-slots.js]: We usually use 0 (or negative values) to denote anons

Comment (JoshuaRogan) [reply app/services/affiliate-slots.js]: I see. Good call

File: app/components/affiliate-unit.js [hunk 3]
```diff
@@ -29,6 +29,12 @@ export default Component.extend(
       } else if (!this.isInContent && this.unit.links && this.unit.links.search) {
         this.unit.link = this.unit.links.search;
       }
+
+      if (this.unit.campaign === 'ddb') {
```
Comment (vforge) [inline app/components/affiliate-unit.js:33]: Looks like a hack - is there any way that we can move those params to slots config? (it doesn't have to be right now)

File: app/components/post-search-results-affiliate-item.js [hunk 4]
```diff
@@ -8,12 +8,18 @@ export default Component.extend({
...
 
-  getUnitLink: computed('unit', 'isInContent', function () {
+  getUnitLink: computed('affiliateUnit', 'isInContent', function () {
     if (this.isInContent && this.affiliateUnit.links && this.affiliateUnit.links.article) {
       this.affiliateUnit.link = this.affiliateUnit.links.article;
     } else if (!this.isInContent && this.affiliateUnit.links && this.affiliateUnit.links.search) {
       this.affiliateUnit.link = this.affiliateUnit.links.search;
     }
+
+    if (this.affiliateUnit.campaign === 'ddb') {
```
Comment (vforge) [inline app/components/post-search-results-affiliate-item.js:18]: Looks like a hack - is there any way that we can move those params to slots config? (it doesn't have to be right now)

File: app/services/affiliate-slots-units.json [hunk 5]
```diff
@@ -1,4 +1,14 @@
 [
+  {
+    "campaign": "ddb",
+    "category": "ddb-test-unit-id",
+    "tagline": "Suggested For You",
+    "image": "https://static.wikia.nocookie.net/2d18111c-bc45-458f-9898-d2d4e3a50589",
+    "heading": "CLICK ME FOR DDB",
+    "subheading": "DDB subheading",
+    "link": "https://www.dndbeyond.com/",
```
Comment (cschucode) [inline app/services/affiliate-slots-units.json:9]: Should we add a dummy links field here?

File: app/components/affiliate-unit.js (hunk 3 above)
Comment (JoshuaRogan) [reply app/components/affiliate-unit.js:33]: How would you see that working?

Comment (JoshuaRogan) [reply app/components/affiliate-unit.js:33]: Do you mean continue what chris was doing by expanding the number of links?

File: app/services/affiliate-slots-units.json (hunk 5 above)
Comment (JoshuaRogan) [reply app/services/affiliate-slots-units.json:9]: Why do you think we should change this?

Comment (cschucode) [reply app/services/affiliate-slots-units.json:9]: To match our existing affiliate units. Or, is there only going to be one link?

Comment (JoshuaRogan) [reply app/services/affiliate-slots-units.json:9]: Oh I see.

**Review by vforge — APPROVED** [review]

File: app/components/post-search-results-affiliate-item.js (hunk 4 above)
Comment (JoshuaRogan) [reply app/components/post-search-results-affiliate-item.js:18]: discussed offline

**Pull request merged**